docs/DeveloperGuide.md

```diff
@@ -18,12 +44,162 @@
+ 
+5. A new `ExitCommand()` is called.
+
+6. `ExitCommand` is returned to parseCommand().
+
+7. `ExitCommand` is returned to main().
+
+8. The Exit Command is now executed via `command.execute()`.
+
+9. `runCommandTillExit()` will now call `command.isExit()` which returns `true`.
+
+10. The loop is now broken and ui will call `ui.showEndingMessage()`.
+
+11. The program will now exit.
+
+<p align="center">
+    <img src="images/ExitCommandSequenceDiagram.png">
+    <br />
+    <i>Figure 2: Sequence Diagram for Bye Command</i>
+</p>
+
+### List Pet Command
+
+The `ListPetCommand` inherits its properties from the abstract `Command` class.
+
+The command to list all pets currently available is `list`. When called, the program prints all the names and statistics of the existing pets (if any), followed by the number of pets currently in the list.
+
+This is how the ListPetCommand works:
+
+1. The `main()` method calls `ui.getUserInput()` from `ui`. The user's command is parsed through `commandParser.parseCommand()`.
+
+2. Within `commandParser.parseCommand()`, `newCommand()` will identify the command received (`list`).
+
+3. `ListPetCommand()` calls the `list()` method from the `PetList` class.
+
+4. `PetList.list()` then iterates through all pets in `petList`, calling `pet.getPetName()`, `pet.getPetType()`, `pet.getAge()` and `pet.getWeight()` each iteration, then prints the number of pets.
+
+5. The loop is now broken to return `ListPetCommand` to `parseCommand()`.
+
+6. `ListPetCommand` is returned to `main()`.
+
+7. The program is now ready to receive another command.
+
+<p align="center">
+    <img src="images/ListPetCommand.png">
+    <br />
+    <i>Figure 3: Sequence Diagram for List Pet Command</i>
+</p>
+
+### [Added] Add Pet Stat Feature
+
+#### Implementation
+
+The add pet stat mechanism is facilitated by the `Pet` class. It is stored internally as a `petList` created under the `PetList` class. Additionally, it  implements the following operations:
+
+- `Pet#addStat()` - Checks which type of stat the user wants to add (i.e. weight, age, pet type)
+- `Pet#setAge()` - Sets the age of the pet
+- `Pet#setPetType()` - Sets the type of the pet (e.g. Dog, Cat, Parrot)
+- `Pet#setWeight()` - Sets the weight of the pet
+
+These operations are exposed in the `PetList` class as `PetList#addStat()` and `AddStatCommand` class as `AddStatCommand#execute()`.
+
+<p align="center">
+    <img src="images/AddPetStatFeatureClassDiagram.png">
+    <br />
+    <i>Figure 4: Class Diagram for Add Pet Stat Feature</i>
+</p>
+
+Given below is an example usage scenario and how the add pet stat mechanism behaves.
+
+Step 1. After the user launches the application and added a pet named "Bob", a `Pet` object will be initialised and saved in the `petList`.
+
+<p align="center">
+    <img src="images/AddPetStatFeatureObjectDiagram (1).png">
+    <br />
+    <i>Figure 5: Object Diagram for Add Pet Stat Feature After Step 1</i>
```

Comment:
Maybe a different name for the list of pet could be used to avoid confusion (petList and PetList)